pkg/download/download_test.go

```diff
@@ -114,3 +120,36 @@ func TestGetAPIList(t *testing.T) {
 	list, err = getAPIList("synthetic-location-test,   extension-test, alerting-profile")
 	assert.ErrorContains(t, err, "There were some errors in the API list provided")
 }
+
+func TestCreateConfigsFromAPI_NoJSONCreated_NoEmptyFolder(t *testing.T) {
+	apiMock := api.CreateAPIMockFactory(t)
+	client := rest.CreateDynatraceClientMockFactory(t)
+	jcreator := jsoncreator.CreateJSONCreatorMock(t)
+	ycreator := yamlcreator.CreateYamlCreatorMock(t)
+	fs := util.CreateTestFileSystem()
+	list := []api.Value{{Id: "d", Name: "namevalue"}}
+
+	client.EXPECT().
+		List(gomock.Any()).Return(list, nil)
+
+	apiMock.EXPECT().
+		GetId().Return("synthetic-monitor").AnyTimes()
+
+	apiMock.EXPECT().
+		IsSingleConfigurationApi().Return(false).AnyTimes()
+
+	jcreator.EXPECT().
+		CreateJSONConfig(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+		Return("demo.json", "demo", true, nil) // every value is filtered
+
+	ycreator.EXPECT().
+		CreateYamlFile(gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
I think `CreateYamlFile` has changed to `WriteYamlFile` in the latest version. Please rebase and update the mock usage :)